codex/utils/trackedfutures.nim

```diff
@@ -42,6 +40,8 @@ proc track*[T, U](future: Future[T], self: U): Future[T] =
 proc cancelTracked*(self: TrackedFutures) {.async.} =
   self.cancelling = true
 
+  trace "cancelling tracked futures"
```

Comment:
I actually think one is a bit helpful without being too noisy. I removed the other one simply because it created too much noise and doesn't contain contextual information (contained in the layer above).